plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -94,7 +296,11 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
                 pattern.containsMatchIn(file.path)
             }
         }
-        deferredResults.any { it.await() }
+
+        val ignoredByPattern = deferredResults.any { it.await() }
+        val ignoredByExtension = !isFileExtensionAllowed(file)
+
+        ignoredByPattern || ignoredByExtension
```

Comment:
do we need return this value?

Reply:
Yes, otherwise the filtering logic won't work